docs/build/guides/conventions/extending-wasm-ttl.mdx

```diff
@@ -0,0 +1,91 @@
+---
+title: Extend a deployed contract’s TTL with code
+description: How to extend the TTL of a deployed contract's Wasm code using JavaScript SDK
+---
+
+# Extending a deployed contract’s TTL using code
+
+When a smart contract is deployed on the Stellar network, its WebAssembly (Wasm) code has a [Time To Live (TTL)](../../../learn/encyclopedia/storage/state-archival.mdx#ttl) that determines how long it remains accessible.
+
+The TTL is the number of ledgers between the current ledger and the final ledger for which the contract data can still be accessed. If the TTL expires, the contract's code becomes archived and inaccessible. To prevent this, you need to periodically extend the TTL of the contract's Wasm code.
+
+This guide will show you how to extend the TTL of a deployed contract's Wasm code using JavaScript.
+
+## Prerequisites
+
+- Stellar SDK: `npm install @stellar/stellar-sdk`
+- A Soroban RPC endpoint (e.g., `https://soroban-testnet.stellar.org`)
+- Basic knowledge of Stellar SDK
+
+## Process overview
+
+- Get the contract's footprint,
+- Set a reasonable resource fee (perhaps start at 10,000 stroops as this is a costly operation),
+- Set Soroban data as read-only,
+- Set the desired ledger TTL to expire the contract,
+- Create an operation `StellarSdk.Operation.extendFootprintTtl`,
+- Note that a resource fee and base fee are both charged in this operation.
+
+## JavaScript code
+
+The code below uses Nodejs environment but same concept can also be applied in the browser using Freighter wallet or using any other [Stellar SDK](/docs/tools/sdks/library).
+
+```javascript
+import * as StellarSdk from "@stellar/stellar-sdk";
+
+import { Server } from "@stellar/stellar-sdk/rpc";
+
+async function extendContractWasmTTL(contractId, sourceKeypair) {
+  const server = new Server("https://soroban-testnet.stellar.org");
+
+  // Create a new transaction builder
+  const account = await server.getAccount(sourceKeypair.publicKey());
+  const fee = "200100"; // Base fee plus resource fee
+
+  // Get the contract instance
+  const contract = new StellarSdk.Contract(contractId);
+  const instance = contract.getFootprint();
+
+  // Set the Soroban data and create an operation to extend the contract's TTL
+  const sorobanData = new StellarSdk.SorobanDataBuilder()
+    .setResourceFee(200_000)
+    .setReadOnly([instance])
+    .build();
```

Comment:
I think it's worthwhile here to describe in a little more detail what the `SorobanDataBuilder()` is doing, and _how_ you're using it. Explain why you're using `setReadOnly()` as opposed to `setReadWrite()`. That kind of stuff.